File: tournaments/sameElementsNaive/sameElementsNaive.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub sameElementsNaive {
    my ($a, $b) = @_;

    my $result = 0;
    my %store = map { $_, 1 } @$a;

    foreach my $item (@$b) {
        if ($store{$item}) {
            ++$result;
        }
    }

    return $result;
}

File: tournaments/sameElementsNaive/sameElementsNaive.js
function sameElementsNaive(a, b) {

    var result = 0;

    for (var i = 0; i < a.length; i++) {
        for (var j = 0; j < b.length; j++) {
            if (a[i] === b[j]) {
                result++;
            }
        }
    }
    return result;
}
